backend/Squirrel.Core/Squirrel.Core.BLL/Services/AuthService.cs

```diff
@@ -3,18 +3,50 @@
 using Squirrel.Core.BLL.Services.Abstract;
 using Squirrel.Core.Common.DTO.Auth;
 using Squirrel.Core.Common.Interfaces;
+using Squirrel.Core.Common.DTO.Users;
 using Squirrel.Core.DAL.Context;
 using Squirrel.Core.DAL.Entities;
+using Google.Apis.Auth;
+using Microsoft.Extensions.Options;
 
 namespace Squirrel.Core.BLL.Services;
 
 public sealed class AuthService : BaseService, IAuthService
 {
     private IJwtFactory _jwtFactory;
+    private readonly string _googleClientId;
     
-    public AuthService(SquirrelCoreContext context, IMapper mapper, IJwtFactory jwtFactory) : base(context, mapper)
+    public AuthService(SquirrelCoreContext context, IMapper mapper, IJwtFactory jwtFactory, IOptions<AuthenticationSettings> authSettings) : base(context, mapper)
     {
-        _jwtFactory = jwtFactory;
+       _jwtFactory = jwtFactory;
+       _googleClientId = authSettings.Value.GoogleClientId;
+    }
+
+    public async Task<AuthUserDTO> AuthorizeWithGoogle(GoogleToken googleToken)
+    {
+        var payload = await GoogleJsonWebSignature.ValidateAsync(googleToken.IdToken, new GoogleJsonWebSignature.ValidationSettings
+        {
+            Audience = new List<string> { _googleClientId }
+        });
+
+        //var userEntity = await _context.Users
+        //    .Include(u => u.Avatar)
+        //    .FirstOrDefaultAsync(u => u.Email == payload.Email);
+        //
+        //if (userEntity == null)
+        //{
+        //    throw new NotFoundException(nameof(User));
+        //}
+        //
+        //var user = _mapper.Map<UserDTO>(userEntity);
+
+        //var token = JwtGenerator.GenerateNewToken(user);
+
+        return new AuthUserDTO
+        {
+            //User = userDTO,
+            //Token = acessTokenDTO
+        };
```

Comment:
why is it commented? if this functionality is to be implemented later, than mark it with TODO

Reply:
yes, it will be implemented later - TODO added